xbmc/cores/AudioEngine/Sinks/AENullSink.cpp

```diff
@@ -0,0 +1,103 @@
+#include "AENullSink.h"
+
+#include "AESinkFactory.h"
+
+#include <algorithm>
+#include <array>
+#include <chrono>
+#include <thread>
+
+using namespace std::chrono_literals;
+
+namespace
+{
+
+constexpr std::chrono::duration<double, std::ratio<1>> DEFAULT_BUFFER_DURATION = 0.200s;
+constexpr int DEFAULT_PERIODS = 4;
+constexpr std::chrono::duration<double, std::ratio<1>> DEFAULT_PERIOD_DURATION =
+    DEFAULT_BUFFER_DURATION / DEFAULT_PERIODS;
+
+} // namespace
+
+bool CAENullSink::Register()
+{
+  AE::AESinkRegEntry entry;
+  entry.sinkName = "NULL";
+  entry.createFunc = CAENullSink::Create;
+  entry.enumerateFunc = CAENullSink::EnumerateDevicesEx;
+  entry.cleanupFunc = CAENullSink::Destroy;
+  AE::CAESinkFactory::RegisterSink(entry);
+
+  return true;
+}
+
+std::unique_ptr<IAESink> CAENullSink::Create(std::string& device, AEAudioFormat& desiredFormat)
+{
+  auto sink = std::make_unique<CAENullSink>();
+  if (sink->Initialize(desiredFormat, device))
+    return sink;
+
+  return {};
+}
+
+void CAENullSink::EnumerateDevicesEx(AEDeviceInfoList& list, bool force)
+{
+  CAEDeviceInfo info;
+  info.m_deviceType = AE_DEVTYPE_PCM;
+  info.m_deviceName = "Default";
+  info.m_displayName = "Default";
+  info.m_displayNameExtra = "Default Output Device (NULL)";
+  info.m_wantsIECPassthrough = true;
```

Comment:
If you want this, you should also add a 16 bit integer format, else you can set it to be pcm only.